Mixpanel/MixpanelInstance.swift

```diff
@@ -686,6 +686,11 @@ extension MixpanelInstance {
      */
     open func reset() {
         trackingQueue.async() {
+            
+            self.networkQueue.sync {
+                return;
```

Comment:
i think you can deadlock here if you call `flush` and `reset` and both are added to the queues before either executes. then you have flush happening on the network queue and reset happening on the tracking queue, with reset only able to complete after the network queue is emptied and flush needing to do a sync operation on the tracking queue so it can't complete